### ISSUE TITLE
False Positive: Replace unused for index with underscore

### ISSUE THREAD

**Issue by hugobuddel** [sourcery-ai/sourcery#49]
### Issue description or question
Sourcery complains about this code with a "Replace unused for index with underscore" at the location marked with "FP here"
```python
def translate(data, keys):
    r = ['abcd']
    nos = 100
    for i in range(nos):  # <- FP here
        r.append([data[k][i] for k in keys])
    return r
```

It converts the code to
```python
def translate(data, keys):
    r = ['abcd']
    nos = 100
    for _ in range(nos):
        r.append([data[k][i] for k in keys])
    return r
```
Which will lead to an exception:
```
NameError: name 'i' is not defined
```

The code is contrived, reducing it further will lead to other suggestions (like use list-comprehension). Nevertheless, Sourcery should not suggest this particular change, because it breaks the code.

### Sourcery Version

Sourcery plugin for PyCharm.

> Plan: Sourcery Open Source. Plugin version: 0.3.7. Sourcery Version: 0.7.0.

### Code editor or IDE name and version

PyCharm 2020.1.1 (Professional Edition)
Build #PY-201.7223.92, built on April 30, 2020
Runtime version: 11.0.6+8-b765.40 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1088M
Cores: 8
Non-Bundled Plugins: mobi.hsz.idea.gitignore, mobi.hsz.idea.latex, nl.rubensten.texifyidea, sourcery.pycharm-plugin, org.sonarlint.idea, org.toml.lang, org.rust.lang

### OS name and version

Windows 10, build 19635.1

**Comment by hugobuddel** [comment]
Shorter:
```python
def translate(data):
    r = ['abcd']
    for i in range(100):  # <- "Replace unused for index with underscore" false positive here
        r.append(data[i] for k in [1, 2])
    return r
```
This function is even more nonsensical.

**Comment by Hellebore** [comment]
Definitely an issue - thanks for taking the time to raise it! 

I will get cracking on fixing it.

**Comment by Hellebore** [comment]
Fix identified - should be in our next release.

**Comment by Hellebore** [comment]
Fixed with vscode 0.2.5 and pycharm 0.4.0 releases.
